### DISCUSSION TITLE
Prisma imports - why are there two directories?

### DISCUSSION THREAD

**Discussion by izaacdb** [prisma/prisma#9155] [Q&A]
If I import from `@prisma/client` my IDE doesn't accept that the enums that are defined there exist. I also get a ton of slow autocompletion, stuff just generally not working very well at all.

If I switch the imports to `.prisma/client` things seem to work fine.

What is the reason for these two directories? `@prisma/client` just seems to do `export * from '.prisma/client'` making it completely redundant?

**Comment by ryands17** [comment]
@izaacdb 👋 

You can have a look at [this section](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/generating-prisma-client#why-is-prisma-client-generated-into-node_modulesprismaclient-by-default) in the docs as to why we opted for this.

> If I import from @prisma/client my IDE doesn't accept that the enums that are defined there exist. I also get a ton of slow autocompletion, stuff just generally not working very well at all.

This should work fine and it does so in VSCode. If you could share a reproduction then I could have a look at it.

**Comment by izaacdb** [comment]
Thanks @ryan - I use jetbrain's webstorm ide, is that supported?

There isn't any information about whether or not to use `.prisma/client`
for imports there. Is there any potential problem from doing it this way?

It seems to work fine when I do it like this, although I haven't done a CI
build yet.

**Reply by janpio** [reply]
It might be fine, but _not_ doing it that way should not be a problem! Can you please open a bug issue and fill all the info it asks for and possibly help us create a reproduction? We want to fix this :/

**Reply by ryands17** [reply]
Yes Webstorm should fetch TypeScript types properly. It would be great if you could create a bug [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) as Jan mentioned 🙂

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
